frontend/src/testing/pushapi-mock.spec.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @author Stefan Dimitrov (stefan.dimitrov@clouway.com).
+ */
+
+describe('PushApi mock should', function () {
+  beforeEach(module('clouway-push'));
+
+  var pushApi, eventName;
+  beforeEach(inject(function (_pushApi_) {
+    pushApi = _pushApi_;
+    eventName = 'test-event';
+  }));
+
+  it('trigger event handling', function () {
+    var wasHandled = false;
+    var eventMessage = '';
+    pushApi.bind(eventName, function (data) {
+      wasHandled = true;
+      eventMessage = data.message;
+    });
+
+    pushApi.handleEvent({message: 'test', event: eventName});
+
+    expect(wasHandled).toBe(true);
+    expect(eventMessage).toEqual('test');
+  });
+
+  it('trigger handling only for the event', function () {
+    var eventMessage = '';
+    pushApi.bind(eventName, function (data) {
+      eventMessage += '1';
+    });
+
+    pushApi.bind('another event', function (data) {
+      eventMessage += '2';
+    });
+
+    pushApi.bind('another event', function (data) {
+      eventMessage += '3';
+    });
+
+    pushApi.bind(eventName, function (data) {
+      eventMessage += '4';
+    });
+
+    pushApi.bind(eventName, function (data) {
+      eventMessage += '5';
+    });
+
+    pushApi.handleEvent({event: eventName});
```

Comment:
not a good naming ` pushApi.handleEvent({event: eventName});` handle event should happens in a handler classes , maybe you should `fire`  that event